src/Fixer/ClassNotation/FinalInternalClassFixer.php

```diff
@@ -177,23 +214,62 @@ private function isClassCandidate(TokensAnalyzer $tokensAnalyzer, Tokens $tokens
             return false; // ignore class; it is abstract or already final
         }
 
-        $docToken = $tokens[$tokens->getPrevNonWhitespace($index)];
+        $decisions = [];
+        $currentIndex = $index;
 
-        if (!$docToken->isGivenKind(T_DOC_COMMENT)) {
-            return $this->configuration['consider_absent_docblock_as_internal_class'];
+        $acceptTypes = [
+            CT::T_ATTRIBUTE_CLOSE,
+            T_DOC_COMMENT,
+            T_COMMENT, // Skip comments
+        ];
+
+        if (\defined('T_READONLY')) {
+            // Skip readonly classes for PHP 8.2+
+            $acceptTypes[] = T_READONLY;
         }
 
-        $doc = new DocBlock($docToken->getContent());
+        while ($currentIndex) {
+            $currentIndex = $tokens->getPrevNonWhitespace($currentIndex);
+
+            if (!$tokens[$currentIndex]->isGivenKind($acceptTypes)) {
+                break;
+            }
+
+            if ($this->checkAttributes && $tokens[$currentIndex]->isGivenKind(CT::T_ATTRIBUTE_CLOSE)) {
+                $attributeStartIndex = $tokens->findBlockStart(Tokens::BLOCK_TYPE_ATTRIBUTE, $currentIndex);
+                $decisions[] = $this->isClassCandidateBasedOnAttribute($tokens, $attributeStartIndex, $currentIndex);
+
+                $currentIndex = $attributeStartIndex;
+            }
+
+            if ($tokens[$currentIndex]->isGivenKind([T_DOC_COMMENT])) {
+                $decisions[] = $this->isClassCandidateBasedOnPhpDoc($tokens, $currentIndex);
+            }
+        }
+
+        if (false !== array_search(false, $decisions, true)) {
+            return false;
+        }
+
+        if (false !== array_search(true, $decisions, true)) {
+            return true;
+        }
+
+        return !\count($decisions) && $this->configuration['consider_absent_docblock_as_internal_class'];
```

Comment:
```suggestion
        if (in_array(false, $decisions, true)) {
            return false;
        }

        return in_array(true, $decisions, true)
            || ([] === $decisions && $this->configuration['consider_absent_docblock_as_internal_class']);
```

- [PHP Inspections (EA Extended)](https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-) plugin for PHPStorm marks it as warning and suggests changing `array_search()` to simple `in_array()` and I agree with it 😉
- I prefer explicit `[] === $foo` rather than `!count()`, and there are several `[] ===` already in the codebase